src/ethereumjs-abi-utils.ts

```diff
@@ -0,0 +1,390 @@
+/* eslint jsdoc/require-description: 0 */
+/* eslint jsdoc/require-returns: 0 */
+/* eslint jsdoc/match-description: 0 */
+/* eslint jsdoc/require-param-description: 0 */
+
+import {
+  toBuffer,
+  setLengthRight,
+  setLengthLeft,
+  isHexPrefixed,
+  zeros,
+} from '@ethereumjs/util';
+import { stripHexPrefix } from 'ethjs-util';
+import BN from 'bn.js';
+import { normalize } from './utils';
+
+//
+// Methods borrowed and somewhat adapted from ethereumjs-abi@0.6.8:
```

Comment:
this is a lot of code - it's not available in a sufficient manner elsewhere?

Reply:
not that I'm seeing... but can look around more!

Reply:
Yeah so for instance I looked into using ethers version of `solidityPack` here https://docs.ethers.io/v5/api/utils/hashing/#utils-solidityKeccak256, which ultimately depends on https://npmfs.com/package/js-sha3/0.8.0/src/sha3.js for its implementation of `Keccak256`

Actually looking at it again, https://www.npmjs.com/package/@ethersproject/solidity might be the way to go to replace `solidityPack`... it has more than we need but would eliminate some of these new utility functions I brought in.

Reply:
using the `keccak256` (which uses `pack`) function from `@ethersproject/solidity` is causing a bunch of issues, some of which I have been able to fix, but unsure if this is a worthwhile path, will bring it to you tomorrow at office hours. Would like to get some clarity about measuring the impact here before proceeding.

Reply:
after review, i think its ok to take on the abi code burden